File: Php/app/Urls/CashgameChartJsonUrlModel.php
namespace app\Urls{

	use app\RouteFormats;
	use app\Urls\BaseClasses\HomegameYearUrlModel;
	use app\UrlFormatter;
	use entities\Homegame;

	class CashgameChartJsonUrlModel extends HomegameYearUrlModel{

		public function __construct(Homegame $homegame, $year){
			parent::__construct(RouteFormats::cashgameChartJson, RouteFormats::cashgameChartJsonWithYear, $homegame, $year);
		}

	}

}

File: Web/Models/UrlModels/CashgameChartJsonUrlModel.cs
using Web.Routing;

namespace Web.Models.UrlModels
{
    public class CashgameChartJsonUrlModel : HomegameWithOptionalYearUrlModel
    {
        public CashgameChartJsonUrlModel(string slug, int? year)
            : base(RouteFormats.CashgameChartJson, RouteFormats.CashgameChartJsonWithYear, slug, year)
        {
        }
    }
}